Client/Frontend/Browser/SearchEngines.swift

```diff
@@ -149,12 +149,18 @@ class SearchEngines {
             let languageWithoutDialect = language.componentsSeparatedByString("-").first!
             searchDirectory = pluginBasePath.stringByAppendingPathComponent(languageWithoutDialect)
             if language == languageWithoutDialect || !NSFileManager.defaultManager().fileExistsAtPath(searchDirectory) {
-                searchDirectory = fallbackDirectory
+                searchDirectory = fallbackDirectory as String
             }
         }
 
-        let index = searchDirectory.stringByAppendingPathComponent("list.txt")
-        let listFile = String(contentsOfFile: index, encoding: NSUTF8StringEncoding, error: &error)
+        let index = (searchDirectory as NSString).stringByAppendingPathComponent("list.txt")
+        let listFile: String?
+        do {
+            listFile = try String(contentsOfFile: index, encoding: NSUTF8StringEncoding)
+        } catch let error1 as NSError {
+            error = error1
+            listFile = nil
+        }
```

Comment:
For handling this error and assertion, we should move away from ever using NSError unless we need to. We can also condense this do statement:

```
        let listFile = try? String(contentsOfFile: index, encoding: NSUTF8StringEncoding)
        assert(listFile != nil, "Failed to read search engine list")
```

Doing this also removes the need for the assertion